### PR TITLE
Adding user code & keymaps for dumbpad & hotdox

### PR DESCRIPTION
**Pull request by imchipwood**
## Description

Adding personal keymaps for dumbpad/v1x & hotdox keyboards

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dumbpad/v1x/keymaps/imchipwood/keymap.c [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include <print.h>
+#include <imchipwood.h>
```
Comment (drashna) [inline keyboards/dumbpad/v1x/keymaps/imchipwood/keymap.c:18]:
```suggestion
#include "imchipwood.h"
```

File: keyboards/dumbpad/v1x/keymaps/imchipwood/keymap.c [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+#define _BASE 0
+#define _SUB 1
```
Comment (drashna) [inline keyboards/dumbpad/v1x/keymaps/imchipwood/keymap.c:21]:
```suggestion
enum custom_layers {
    _BASE,
    _SUB,
};
```

File: keyboards/hotdox/keymaps/imchipwood/keymap.c [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "imchipwood.h"
+
+// Layer names
+#define _QWERTY     0 // default layer
+#define _NUM    1 // numbers/etc.
+#define _MED      2 // media keys
```
Comment (drashna) [inline keyboards/hotdox/keymaps/imchipwood/keymap.c:22]: could you convert this to an enum, as well?

File: users/imchipwood/imchipwood.c [hunk 4]
```diff
@@ -0,0 +1,126 @@
+#include "imchipwood.h"
```
Comment (drashna) [inline users/imchipwood/imchipwood.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/imchipwood/imchipwood.c [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+static td_state_t td_state;
```
Comment (drashna) [inline users/imchipwood/imchipwood.c:3]: Heads up, if you have more than one dance using this, .... don't. 

You want one struct for each dance.  So, 3 from the looks of it.  

```suggestion
static td_state_t td_state[3];
```

File: users/imchipwood/imchipwood.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
+#ifndef USERSPACE
+#define USERSPACE
```
Comment (drashna) [inline users/imchipwood/imchipwood.h:2]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#pragma once
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: users/imchipwood/imchipwood.c (hunk 5 above)
Comment (imchipwood) [reply users/imchipwood/imchipwood.c:3]: Got it - thanks!

File: users/imchipwood/imchipwood.c (hunk 4 above)
Comment (imchipwood) [reply users/imchipwood/imchipwood.c:16]: Done.

File: keyboards/hotdox/keymaps/imchipwood/keymap.c (hunk 3 above)
Comment (imchipwood) [reply keyboards/hotdox/keymaps/imchipwood/keymap.c]: Done.

**Comment by imchipwood** [comment]
Hey - this has sat for quite some time... Anything needed from my side?

File: keyboards/dumbpad/v1x/keymaps/imchipwood/keymap.c [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include <print.h>
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x/keymaps/imchipwood/keymap.c:17]:
```suggestion
```

File: keyboards/dumbpad/v1x/keymaps/imchipwood/keymap.c [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+                   TD(ALT_F7),          KC_F8,          KC_F9, TD(DEL_NLCK),
+                        KC_F4,     TD(CTL_F5),          KC_F6,  KC_KP_MINUS,
+                        KC_F1,     TD(ALT_F2),          KC_F3,  KC_KP_SLASH,
+        KC_MUTE,      _______,  OSM(MOD_LSFT),  OSM(MOD_LCTL),  KC_KP_EQUAL
+    )
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        switch (biton32(layer_state)) {
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x/keymaps/imchipwood/keymap.c:66]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by imchipwood** [comment]
All requested changes are implemented, and the changes have been verified to be working on my own keyboards.

Please re-review or approve and merge! 

Thanks!
Chip

**Review by drashna — APPROVED** [review]
Thanks!

**Pull request merged**